modules/ixBidAdapter.js

```diff
@@ -92,12 +99,73 @@ function bidToImp(bid) {
     imp.ext.sid = `${bid.params.size[0]}x${bid.params.size[1]}`;
   }
 
-  if (bid.params.hasOwnProperty('bidFloor') && bid.params.hasOwnProperty('bidFloorCur')) {
-    imp.bidfloor = bid.params.bidFloor;
-    imp.bidfloorcur = bid.params.bidFloorCur;
+  return imp;
+}
+
+/**
+ * Gets priceFloors floors and IX adapter floors,
+ * Validates and sets the higher one on the impression
+ * @param  {object}    bid bid object
+ * @param  {object}    imp impression object
+ * @param  {string}    mediaType the impression ad type, one of the SUPPORTED_AD_TYPES
+ */
+function _applyFloor(bid, imp, mediaType) {
+  let adapterFloor = null;
+  let moduleFloor = null;
+
+  if (bid.params.bidFloor && bid.params.bidFloorCur) {
+    adapterFloor = { floor: bid.params.bidFloor, currency: bid.params.bidFloorCur };
   }
 
-  return imp;
+  if (utils.isFn(bid.getFloor)) {
+    let _mediaType = '*';
+    let _size = '*';
+
+    if (mediaType && utils.contains(SUPPORTED_AD_TYPES, mediaType)) {
+      const { w: width, h: height } = imp[mediaType];
+      _mediaType = mediaType;
+      _size = [width, height];
+    }
+    try {
+      moduleFloor = bid.getFloor({
+        mediaType: _mediaType,
+        size: _size
+      });
+    } catch (err) {
+      // continue with no module floors
+      utils.logWarn('priceFloors module call getFloor failed, error : ', err);
+    }
+  }
+
+  if (adapterFloor && moduleFloor) {
+    if (adapterFloor.currency !== moduleFloor.currency) {
+      utils.logWarn('The bid floor currency mismatch between IX params and priceFloors module config');
+      return;
+    }
+
+    if (adapterFloor.floor > moduleFloor.floor) {
+      imp.bidfloor = adapterFloor.floor;
+      imp.bidfloorcur = adapterFloor.currency;
+      imp.ext.fl = FLOOR_SOURCE.IX;
+    } else {
+      imp.bidfloor = moduleFloor.floor;
+      imp.bidfloorcur = moduleFloor.currency;
+      imp.ext.fl = FLOOR_SOURCE.PBJS;
+    }
+    return;
+  }
+
+  if (moduleFloor) {
+    imp.bidfloor = moduleFloor.floor;
+    imp.bidfloorcur = moduleFloor.currency;
+    imp.ext.fl = FLOOR_SOURCE.PBJS;
+  } else if (adapterFloor) {
+    imp.bidfloor = adapterFloor.floor;
+    imp.bidfloorcur = adapterFloor.currency;
+    imp.ext.fl = FLOOR_SOURCE.IX;
+  } else {
+    utils.logInfo('No floors available, no floors applied')
```

Comment:
Can you add a prefix to this so when looking at logs it is clear where it is coming from:

utils.logInfo('IX Bid Adapter: No floors available, no floors applied')

Reply:
Added it